app/knights/battle.py

```diff
@@ -0,0 +1,47 @@
+from typing import Dict, Any
+from app.knights.knight import Knight
+
+
+def battle(knights_config: Dict[str, Dict[str, Any]]) -> Dict[str, int]:
+    try:
+        knights = {
+            name: Knight(**data)
+            for name, data in knights_config.items()}
+    except KeyError as e:
+        raise ValueError(f"Missing expected knight data for {str(e)}")
+
+    lancelot = knights.get("lancelot")
+    mordred = knights.get("mordred")
+    arthur = knights.get("arthur")
+    red_knight = knights.get("red_knight")
+
+    if not all([lancelot, mordred, arthur, red_knight]):
+        raise ValueError(
+            "One or more knights are missing in the configuration."
+        )
```

Comment:
Consider improving the error message to specify which knight is missing. This will make debugging easier.